sql/analyzer/rules_test.go

```diff
@@ -396,6 +396,128 @@ func (t *pushdownProjectionAndFiltersTable) TransformUp(f func(sql.Node) (sql.No
 	return f(&pushdownProjectionAndFiltersTable{table.(sql.Table)})
 }
 
+func TestCastCompares(t *testing.T) {
+	f := getRule("cast_compares")
+
+	tests := []struct {
+		name     string
+		node     sql.Node
+		expected sql.Node
+	}{
+		{
+			name: `number >= 2`,
+			node: plan.NewProject(
+				[]sql.Expression{
+					expression.NewGreaterThanOrEqual(
+						expression.NewGetField(1, sql.Int32, "number", false),
+						expression.NewLiteral(int64(2), sql.Int64),
+					),
+				},
+				plan.NewUnresolvedTable("foo"),
+			),
+			expected: plan.NewProject(
+				[]sql.Expression{
+					expression.NewGreaterThanOrEqual(
+						expression.NewConvert(
```

Comment:
The auto casting should be always only on the literal side I think

Reply:
As explained up there in the other comment, the cast can't be just for literals and should be for both expressions because of the "types" the convert expression accepts.

Anyway, I'm moving all the logic to Comparison expression and removing the rule as @erizocosmico suggested.